pkg/conversion/decode.go

```diff
@@ -80,6 +80,10 @@ func (s *Scheme) Decode(data []byte) (interface{}, error) {
 // If obj's version doesn't match that in data, an attempt will be made to convert
 // data into obj's version.
 func (s *Scheme) DecodeInto(data []byte, obj interface{}) error {
+	return s.DecodeIntoWithSpecifiedVersionKind(data, obj, "", "")
+}
+
+func (s *Scheme) DecodeIntoWithSpecifiedVersionKind(data []byte, obj interface{}, specifiedVersion, specifiedKind string) error {
```

Comment:
Please add godoc comment; are specifiedVersion and specifiedKind allowed to be ""? (I suggest disallowing this, but up to you.)